### DISCUSSION TITLE
Feature Request: Categories (means of transportation/modes)

### DISCUSSION THREAD

**Discussion by tyrrminal** [Freika/dawarich#218] [Ideas]
The way I would like to use Dawarich is to display heatmaps of my outdoor fitness activities (running/cycling/walking/kayaking/etc). Currently I used Google Earth for this purpose, but Dawarich would make a great many things better than that. Frequently I want the ability to only see my runs on the map, or only see my rides -- in Google Earth, I use folders for this, to easily toggle on/off large groups of GPX tracks, and furthermore, I have color-coded the tracks so that e.g., runs are red lines and bike rides are blue lines.

Before I would be able to convert from Google Earth to using Dawarich for this, I would need comparable functionality: assigning categories to points/tracks/imports, selecting the color of the point/line that is displayed on the map for a given category, and filtering to select one or more categories to display on the map at a time

**Comment by radh21301** [comment]
I agree. This would be pretty cool.

My use case would be to track my hikes, and bike rides - but still the same concept.

I use overland app for IOS. And I think it has the ability to send information on what kind of activity it is (car, bike, walk) etc..maybe that information can be used? And the user can choose a set of points on the 'Points' page, and assign categories to them. Then on the 'Map' page, we can filter the points based on these categories. Would be pretty handy!

**Comment by eng3** [comment]
I fly alot, about 60 flights a year, 100000mi per year.  As a hobby, I record a GPX track of every flight.
I've contemplated whether I should upload those into here.  It would be nice if I could mark them all into a group/category called "flight" and toggle it on/off.   Ofcourse google also manages to record my location along flights sometimes too so some show up there too so eitherway I have a mess.

**Comment by aahmozart** [comment]
This is pretty cool. Willing to donate $100 to prioritize it @Freika

**Comment by js94x** [comment]
I also use Dawarich to display my outdoor fitness activities. As a simple (if possible for you) workaround I just created three single accounts for each activity. 

One account gets my phone data from Home Assistant. Another one receives gpx from my smartwatch, and the last receives gpx from my bicycle rides (exported automatically from Komoot).

**Comment by GegudeBR** [comment]
This feature would be amazing

**Comment by aahmozart** [comment]
Couldn't Dawarich in theory infer category of transportation (walking, running, flying, driving) from the points themselves based on speed and other props? I guess it won't be 100% accurate but it would suffice for most users out there

**Reply by Freika** [reply]
It could, and I want to work on it! Just don't have enough time for it right now :)

**Reply by madindehead** [reply]
From what I can see, Dawarich is currently trying to do this. I don't understand why it thinks I'm going from Driving to a Motorbike, to sometimes Airplane (although that seems to be jumps in GPS points - maybe I need to have better tracking, although the app is on 'balanced' tracking. 

It's not doing the best job, so in some ways it would be better if it didn't infer and you had to tell it. There's too many 'smart' features in projects that just don't work that well and cause people more work. Right now, this is sadly one of them.

Just my two cents as a casual user.

**Comment by eng3** [comment]
well if there was a way I could upload those an manually mark them as "flight" it wouldn't be a "smart" feature.

But for something like an airplane, it is pretty easy to go by speed.  That is fairly reliable for a GPS as it is a big difference.  I could see why using something else like altitude wouldn't work well.  It's all heuristics at that point.  I have a workaround now where when I fly, and start the track recording, it auto triggers via tasker gpslogger to switch to a different profile.

**Reply by madindehead** [reply]
This makes sense yes.

I'm mainly talking about the current implementation and for me it seems to be very inconsistent.

It seems to think I'm cycling, on a motorcycle, and driving in a car all within the same journey.
